sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,527 @@
+- `asset`: `string` address, Stacks c32-encoded, with contract name suffix
+- `amount`: `number | string` BigInt constructor compatible value
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+### Method `stx_transferNft`
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `asset`: `string` address, Stacks c32-encoded, with contract name suffix
+- `assetId`: `ClarityValue`
+
+`where`
+
+- `ClarityValue`: `string | object` hex-encoded or JSON representation
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+### Method `stx_callContract`
+
+`params`
+
+- `contract`: `string.string` address with contract name suffix, Stacks c32-encoded
+- `functionName`: `string`
+- `functionArgs`: `ClarityValue[]`, defaults to `[]`
+
+`where`
+
+- `ClarityValue`: `string | object` hex-encoded or JSON representation
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+### Method `stx_deployContract`
+
+`params`
+
+- `name`: `string`
+- `clarityCode`: `string` Clarity contract code
+- `clarityVersion?`: `number`
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+### Method `stx_signTransaction`
+
+`params`
+
+- `transaction`: `string` hex-encoded raw transaction
+
+`result`
+
+- `transaction`: `string` hex-encoded raw transaction (signed)
+
+### Method `stx_signMessage`
+
+`params`
+
+- `message`: `string`
+
+`result`
+
+- `signature`: `string` hex-encoded
+- `publicKey`: `string` hex-encoded
+
+### Method `stx_signStructuredMessage`
+
+`params`
+
+- `message`: `string` Clarity value, hex-encoded
+- `domain`: `string` hex-encoded (defined by SIP-018)
+
+`result`
+
+- `signature`: `string` hex-encoded
+- `publicKey`: `string` hex-encoded
+
+### Method `stx_getAddresses`
+
+`result`
+
+- `addresses`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+
+### Method `stx_getAccounts`
+
+> **Comment**: This method is similar to `stx_getAddresses`.
+> It was added to provide better backwards compatibility for applications using Gaia.
+
+`result`
+
+- `accounts`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+  - `gaiaHubUrl`: `string` URL
+  - `gaiaAppKey`: `string` hex-encoded
+
+### Method `stx_updateProfile`
+
+`params`
+
+- `profile`: `object` Schema.org Person object
+
+`result`
+
+- `profile`: `object` updated Schema.org Person object
+
+## Listeners
+
+In addition to the request interface, event listeners may be provided via the `.listen` method.
+Wallets may provide a `.unlisten` method to remove listeners.
+
+- `provider.listen(event: string, listener: (...args: any[]) => void): void`
+- `provider.unlisten(event: string, listener: (...args: any[]) => void): void`
+
+### Event `accountsChanged`
```

Comment:
I think we should avoid `-ed` past tense naming. Browser for instance uses `click` events not `onclicked`.

`onNetworkChange`, `accountChanged`?

(Unsure about `on` prefix)

Reply:
I think the events names are closer to nouns than verbs, e.g.,

- A `click` happened
- A `blur` happened
- An `accountChange` happened

hence no -ed. 

I see the `on` prefix used for the event handlers, e.g., `onmousedown="..."`, not for event names. [Some HTML spec event names](https://html.spec.whatwg.org/multipage/webappapis.html#handler-onclick) for inspiration :)